azure/resource_groups/ProjectCore/template.json

```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "resourceNamePrefix": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]"
+    }
+  },
+  "variables": {
+    "vNetName": "[concat(parameters('resourceNamePrefix'), '-vn')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Network/virtualNetworks",
+      "apiVersion": "2019-07-01",
+      "name": "[variables('vNetName')]",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": ["10.211.0.0/26"]
+        },
+        "subnets": [
+          {
+            "name": "default",
+            "properties": {
+              "addressPrefix": "10.211.0.0/27",
+              "serviceEndpoints": [],
+              "delegations": [],
+              "privateEndpointNetworkPolicies": "Enabled",
+              "privateLinkServiceNetworkPolicies": "Enabled"
```

Comment:
I'd recommend using service endpoints and delegations here. Private link has only just come out in preview. I'd probably avoid this until it's GA.

The block template above allows you to pass in a subnet configuration object. You can include service endpoint and delegation config within that object.

For ref: https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2019-04-01/virtualnetworks/subnets#subnetpropertiesformat-object

Reply:
Yeah, we decided to leave those out to be added when we need them for specific uses.